compiler/zrc_codegen/src/stmt.rs

```diff
@@ -648,6 +648,192 @@ pub(crate) fn cg_block<'ctx, 'input, 'a>(
 
                     Some(exit)
                 }
+
+                TypedStmtKind::InlineAsm {
+                    template,
+                    outputs,
+                    inputs,
+                    clobbers,
+                } => {
+                    // Extract string literals from the template and constraints
+                    use inkwell::types::BasicType;
+                    use zrc_typeck::tast::expr::TypedExprKind;
+
+                    let template_str =
+                        if let TypedExprKind::StringLiteral(string) = template.kind.value() {
+                            string.to_string()
+                        } else {
+                            panic!("inline assembly template must be a string literal");
+                        };
+
+                    // Build constraint string and collect operand information
+                    let mut constraint_parts = Vec::new();
+                    let mut output_places = Vec::new();
+                    let mut output_types = Vec::new();
+                    let mut input_values = Vec::new();
+
+                    // Process output constraints and get their places
+                    for output in &outputs {
+                        let constraint_str = if let TypedExprKind::StringLiteral(string) =
+                            output.constraint.kind.value()
+                        {
+                            string.to_string()
+                        } else {
+                            panic!("inline assembly constraint must be a string literal");
+                        };
+                        constraint_parts.push(constraint_str);
+
+                        // Convert output expression to a place to get its pointer
+                        // The output expression should be an Identifier (variable reference)
+                        let expr_cg = BlockCtx::new(cg, &scope, lexical_block);
```

Comment:
Why are we generating an entirely new lexical block here? Don't we exist in the context of one already?